### PR TITLE
feat: [CDS-77527]: GitX Webhooks Listing Page Details Rows and API integrations , Create Webhook Modal Loading + Success Screens + API Integrations

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Refactored the webhook listing page to independent component rather than dependent on `PageTemplate`
- Webhook lIsting page details row setup and integrating the list API.
- Create Webhook Modal API integrate and success and loading screens added

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/ead8eed3-8378-458f-a65c-c439e5f640a1

https://github.com/harness/harness-core-ui/assets/106532291/c85a721f-dc1a-4d51-b97b-a4a8e8868296

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/27-platform/default-settings/components/EULAComponents.tsx [hunk 1]
```diff
@@ -85,7 +85,7 @@ export const AIDASettingsRenderer = (props: SettingRendererProps): React.ReactEl
                   content: { signed: isSigned }
                 } = await validateEulaSign({
                   queryParams: {
-                    'agreement-type': AidaAgreementType
+                    agreement_type: AidaAgreementType
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/default-settings/components/EULAComponents.tsx:88]: Did the param casing change as well? @namanharness Please confirm this change.

File: src/modules/75-cd/pages/webhooks/Webhooks.tsx [hunk 2]
```diff
@@ -1,142 +1,211 @@
...
+      limit: size,
+      page: page ? page - 1 : 0
+    }
+  })
+
+  enum STATUS {
+    'loading',
+    'error',
+    'ok'
+  }
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/Webhooks.tsx:72]: nit: declare outside the function

File: src/modules/75-cd/pages/webhooks/WebhooksList/WebhooksListColumns.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+      // disabled={ TODO handle the disable case here }
+      onChange={() => {
+        // handle the enabled webhook onChange here
+      }}
+    />
+  )
+}
+
+export function WebhookMenu({
+  webhook_identifier,
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/WebhooksList/WebhooksListColumns.tsx:136]: nit: We can maintain camel casing for our code variables

**Review by PratyushGar — APPROVED** [review]

File: src/modules/75-cd/pages/webhooks/Webhooks.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/Webhooks.tsx:72]: will do in next pr

File: src/modules/75-cd/pages/webhooks/WebhooksList/WebhooksListColumns.tsx (hunk 3 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhooksList/WebhooksListColumns.tsx:136]: will do in next pr

Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhooksList/WebhooksListColumns.tsx:136]: will do in next pr

File: src/modules/27-platform/default-settings/components/EULAComponents.tsx (hunk 1 above)
Comment (namanharness) [reply src/modules/27-platform/default-settings/components/EULAComponents.tsx:88]: https://github.com/harness/harness-core-ui/assets/110456765/9f97511b-23c5-4703-9f77-58d43a62db3c

@vikrant-harness @mayankVermaHarness  the new casing for validating EULA is not working

**Review by namanharness — APPROVED** [review]

**Comment by vikrant-harness** [comment]
The coverage will be taken up as a part of separate ticket , 1 Lint warning cannot be resolved at the moment will do in subsequent PR

File: src/modules/27-platform/default-settings/components/EULAComponents.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/27-platform/default-settings/components/EULAComponents.tsx:88]: Discussed offline this should work

**Pull request merged**